WeatherApp/WeatherApp.Webpage/Services/CurrentWeatherService.cs

```diff
@@ -1,57 +1,53 @@
-﻿using Microsoft.AspNetCore.Hosting;
+﻿using Microsoft.Extensions.Configuration;
 using Newtonsoft.Json.Linq;
 using System;
 using System.Net.Http;
+using System.Threading.Tasks;
 using WeatherApp.WebSite.Models;
 using WeatherApp.WebSite.Services.Interfaces;
 
 namespace WeatherApp.WebSite.Services
 {
     public class CurrentWeatherService : ICurrentWeatherService
     {
-        const string API_KEY = "3c850b0463346d2fffad82b66d5eb561";
+        readonly string _apiKey;
+        readonly string _baseUrl;
 
-        public CurrentWeatherService(IWebHostEnvironment webHostEnvironment)
+        public HttpClient Client { get; set; }
+
+        public CurrentWeatherService(IConfiguration configuration, HttpClient client)
         {
-            WebHostEnvironment = webHostEnvironment;
-        }
+            _apiKey = configuration["WeatherApp:ServiceApiKeys:Openweathermap"];
+            _baseUrl = configuration["ApiBaseUrls:CurrentWeather"];
 
-        public IWebHostEnvironment WebHostEnvironment { get; }
+            client.BaseAddress = new Uri(_baseUrl);
+            Client = client;
+        }
 
-        public CurrentWeather GetCurrentWeather(string city)
+        public async Task<CurrentWeather> GetCurrentWeatherAsync(string city)

```

Comment:
In its current form this is unnecessarily duplicate the word `Current`. I can imagine the following 2 versions:

## Rename method

```csharp
public class CurrentWeatherService 
{
  public async Task<Weather> GetAsync(string city){...}
}
```

## Rename class

```csharp
public class WeatherService 
{
  public async Task<Weather> GetCurrentWeatherAsync(string city){...}
}
```